### DISCUSSION TITLE
Possiblity to pass custom argument to setup.py when installing a package

### DISCUSSION THREAD

**Discussion by mgreive** [pypa/setuptools#3720] [Q&A] [answered]
It used to be possible to pass custom options with pip install that could be accesed in setup.py as described for example in [this ](https://stackoverflow.com/questions/47210058/passing-command-line-arguments-to-pip-install?noredirect=1&lq=1)stackoverflow post. Since --install-options is now deprectated ( https://github.com/pypa/pip/issues/11358) this is not possible anymore.

When trying to use --install-options pip suggests --config-settings settings as a workaround. Is it possible to access the value-key pair specified with --config-setting from within setup.py?

I posted a question regarding this on [stackoverflow](https://stackoverflow.com/questions/74571952/how-to-access-python-config-settings-from-a-python-script-how-to-use-the-pip) but I am starting to wonder if it is even possible to get the result I want with --config-settings. If this is not the case, does setuptools offer an alternative?

**Comment by AlexanderSerov** [comment, marked as answer]
The same [question](https://discuss.python.org/t/passing-command-line-arguments-to-pip-install-after-install-options-deprecation/22981/1)
